app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt

```diff
@@ -0,0 +1,190 @@
+import com.github.se.cyrcle.ui.gambling.GamblingScreen
+import com.github.se.cyrcle.ui.gambling.WheelView
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class GamblingScreenTest {
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Test
+  fun gambling_screen_shows_all_elements() {
+    composeTestRule.setContent { GamblingScreen() }
+
+    composeTestRule.onNodeWithTag("gambling_screen").assertExists()
+    composeTestRule.onNodeWithTag("wheel_canvas").assertExists()
+    composeTestRule.onNodeWithTag("spin_button").assertExists()
+  }
+
+  @Test
+  fun verify_wheel_segments() {
+    composeTestRule.setContent { GamblingScreen() }
+
+    // Verify all segments exist
+    for (i in 0..4) {
+      composeTestRule.onNodeWithTag("segment_$i").assertExists()
+      composeTestRule.onNodeWithTag("segment_text_$i").assertExists()
+    }
+  }
+
+  @Test
+  fun spin_button_triggers_wheel_spin() {
+    composeTestRule.setContent { GamblingScreen() }
+
+    composeTestRule.onNodeWithTag("spin_button").performClick()
+    // Wait for spin animation to start
+    composeTestRule.waitForIdle()
+
+    // Verify wheel is spinning (through visual state or properties)
+    composeTestRule.onNodeWithTag("wheel_canvas").assertExists()
+  }
+
+  @Test
+  fun verify_spin_button_properties() {
+    composeTestRule.setContent { GamblingScreen() }
+
+    composeTestRule
+        .onNodeWithTag("spin_button")
+        .assertHasClickAction()
+        .assertTextEquals("SPIN")
+        .assertHeightIsEqualTo(90.dp)
+        .assertWidthIsEqualTo(90.dp)
+  }
+
+  @Test
+  fun verify_wheel_dimensions() {
+    composeTestRule.setContent { GamblingScreen() }
+
+    composeTestRule
+        .onNodeWithTag("wheel_canvas")
+        .assertHeightIsEqualTo(300.dp)
+        .assertWidthIsEqualTo(300.dp)
+  }
+
+  @Test
+  fun verify_segment_probabilities() {
+    var wheelSpinFunction: (() -> Unit)? = null
+
+    composeTestRule.setContent {
+      WheelView(modifier = Modifier.testTag("wheel_view")).let { spinFn ->
+        wheelSpinFunction = spinFn
+      }
+    }
+
+    // Simulate multiple spins and verify probability distribution
+    repeat(100) {
+      wheelSpinFunction?.invoke()
+      composeTestRule.waitForIdle()
+      // Add delay between spins
+      runBlocking { delay(100) }
+    }
+  }
+
+  @Test
+  fun verify_near_miss_mechanics() {
+    var wheelSpinFunction: (() -> Unit)? = null
+
+    composeTestRule.setContent {
+      WheelView(modifier = Modifier.testTag("wheel_view")).let { spinFn ->
+        wheelSpinFunction = spinFn
+      }
+    }
+
+    // Trigger spins and verify near miss logs
+    repeat(10) {
+      wheelSpinFunction?.invoke()
+      composeTestRule.waitForIdle()
+      runBlocking { delay(100) }
+    }
+  }
+
+  @Test
+  fun verify_spin_animation_completion() {
```

Comment:
Here this test takes up 10 seconds... Would it be possible to add a `composeTestRule.mainClock.advanceTimeBy(10000) like you did in the last test?